Packs/GmailSingleUser/Integrations/GmailSingleUser/GmailSingleUser.py

```diff
@@ -49,11 +50,67 @@
     'Accept': 'application/json',
 }
 
-
+EXECUTION_METRICS = ExecutionMetrics()
 ''' HELPER FUNCTIONS '''
 
 
+def execute_gmail_action(service, action: str, action_kwargs: dict) -> dict:
+    """Executes a specified action on the Gmail API.
+
+    This function dynamically executes an action such as sending an email, retrieving an email,
+    getting attachments, or listing emails based on the specified action and its arguments.
+
+    Args:
+        service: The Gmail API service instance.
+        action (str): The action to perform. Supported actions are "send", "get",
+                      "get_attachments", and "list".
+        action_kwargs (dict): The keyword arguments required for the specified action.
+
+    Returns:
+        dict: The result from the Gmail API call.
+
+    Raises:
+        HttpError: If an error occurs from the Gmail API request.
+        Exception: If a general error occurs during the function execution.
+
+    Note:
+        This function updates execution metrics counters based on the outcome of the API call.
+    """
+    try:
+        match action:
+            case "send":
+                return service.users().messages().send(**action_kwargs).execute()
```

Comment:
why is this one `return` and the others save a `result?` does it have no (interesting) return value?

Reply:
My mistake i will fix it...